scripts/convert_greenbay_yaml.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+"""This source codifies our deprecations and argument modifications to the
+Greenbay.yaml format. It takes a single argument: the path to the greenbay.yaml and will create a redalert.yaml from it."""
+
+import sys
+from collections import OrderedDict
+
+import pyaml
+import yaml
+
+
+# Make it so yaml multi line strings use the | syntax that we prefer
+def str_presenter(dumper, data):
+    data = data.rstrip()
+    if len(data.splitlines()) > 1:  # check for multiline string
+        return dumper.represent_scalar(
+            'tag:yaml.org,2002:str', data, style='|')
+    if data.startswith('{') or data.endswith('}'):
+        return dumper.represent_scalar(
+            'tag:yaml.org,2002:str', data, style="'")
+    return dumper.represent_scalar('tag:yaml.org,2002:str', data, style='')
+
+
+pyaml.add_representer(str, str_presenter)
+
+
+def represent_ordereddict(dumper, data):
+    value = []
+
+    if 'name' in data and 'suites' in data:
+        for key in ['name', 'suites', 'type', 'args']:
+            value.append((dumper.represent_data(key),
+                          dumper.represent_data(data[key])))
+    else:
+        for item_key, item_value in data.items():
+            node_key = dumper.represent_data(item_key)
+            node_value = dumper.represent_data(item_value)
+            value.append((node_key, node_value))
+
+    return yaml.nodes.MappingNode(u'tag:yaml.org,2002:map', value)
+
+
+pyaml.add_representer(OrderedDict, represent_ordereddict)
+pyaml.add_representer(dict, represent_ordereddict)
+
+
+def convert_shell_operation(test):
+    """shell-operation superseded by run-bash-script"""
+    cmd = test['args'].pop('command')
+    test['args']['source'] = cmd.strip()
+    test['type'] = 'run-bash-script'
+    return test
+
+
+def convert_run_program_system_python(test):
+    """run-program-system-python* superseded by run-python-script"""
+    if 'windows' in test['name'] and test['type'].endswith('python2'):
+        test['args']['interpreter'] = 'C:\\Python27\\python.exe'
+    elif 'windows' in test['name']:
+        test['args']['interpreter'] = 'C:\\Python36\\python.exe'
+    elif test['type'].endswith('python2'):
+        test['args']['interpreter'] = 'python2'
+    test['type'] = 'run-python-source'
```

Comment:
Should this be 'run-python-script' instead?

Reply:
Correct